lib/trino-ignite-patched/src/main/java/org/apache/ignite/shaded/internal/util/GridUnsafe.java

```diff
@@ -0,0 +1,2281 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.shaded.internal.util;
+
+import org.apache.ignite.IgniteSystemProperties;
+import org.apache.ignite.internal.util.DirectBufferCleaner;
+import org.apache.ignite.internal.util.FeatureChecker;
+import org.apache.ignite.internal.util.ReflectiveDirectBufferCleaner;
+import org.apache.ignite.internal.util.UnsafeDirectBufferCleaner;
+import org.apache.ignite.internal.util.typedef.internal.A;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import sun.misc.Unsafe;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.nio.Buffer;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import java.security.PrivilegedActionException;
+import java.security.PrivilegedExceptionAction;
+
+import static com.google.common.base.Preconditions.checkState;
```

Comment:
@wendigo I used this instead of `assert` since checkstyle forbids it. I also formatted the whole file.